File: Segundo Periodo/AED2/TPs/PalindromoRecursivo.c
#include <stdio.h>
#include <string.h>

int comp(char * x, char * y,int i){
    if(i == strlen(x)){
        return 1;
    }
    if(x[i] != y[i]){
        return 0;
    }
    return comp(x,y,i+1);
}

int strcomparar(char * x, char * y){
    if(strlen(x) != strlen(y)){
        return 0;
    }else{
        return comp(x,y,0);
    }
}

int isPalindromo(char * x, int i){
    if((int)(strlen(x)/2) == i){
        return 1;
    }
    if(x[i] != x[strlen(x)-i-1]){
        return 0;
    }
    return isPalindromo(x,i+1);
}

void loop(){
    char x[255] = "";

    fgets(x,sizeof(x),stdin);
    x[strlen(x)-1] = '\0';
    if(!strcomparar(x,"FIM")){
        if(strcomparar(x,"FIM")){
            return;
        }
        if(isPalindromo(x,0)){
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }
        loop();
    }
}

int main(){
    loop();
    return 0;
}

File: Segundo Periodo/AED2/TPs/PalindromoRecursivo.java
public class PalindromoRecursivo {

    public static boolean isEquals(String x, String y, int i){
        if(i == x.length()){
            return true;
        }
        if(x.charAt(i) != y.charAt(i)){
            return false;
        }
        return isEquals(x, y,i+1);
    }
    public static boolean isEquals(String x, String y){
        if(x.length() == y.length()){
            return isEquals(x, y,0);
        }else{
            return false;
        }
    }
    
    public static boolean isPalindromo(String x,int i){
        if(i == (int)(x.length())/2){
            return true;
        }
        if(x.charAt(i) != x.charAt(x.length()-1-i)){
            return false;
        }
        return isPalindromo(x, i+1);
    }
    
    public static void startLoop(){
        String x = "";
        if(!isEquals(x, "FIM")){
            x = MyIO.readLine();
            if(isEquals(x, "FIM")){
                return;
            }
            if(isPalindromo(x, 0)){
                System.out.println("SIM");
            }else{
                System.out.println("NAO");
            }
            startLoop();
        }
    }

    public static void main(String[] args) {
        startLoop();
    }
}
